Tools/NTupleReader.h

```diff
@@ -43,20 +43,44 @@
 
 class NTupleReader;
 
+//"Iterator" to allow use with for loops
+class NTupleReaderIterator
```

Comment:
What will you use this class for?

Reply:
This class is the internal mechanism that allows the following line of code to work.  Not really needed, but I thought it was fun once I realized how easy it was to do.  
```
NTupleReader tr(tree);
for(auto& event : tr)
{
    auto& jet_pt = event.getVec<float>("Jet_pt");
}
```